test/studies/lcals/LCALSDataTypes.chpl

```diff
@@ -0,0 +1,280 @@
+module LCALSDataTypes {
+  use LCALSParams;
+  use LCALSEnums;
+
+  class vector {
+    type eltType;
+    var D = {0..-1};
+    var A: [D] eltType;
+    proc this(i: int) ref {
+      return A[i];
+    }
+    proc push_back(e: eltType) {
+      A.push_back(e);
+    }
+    proc size() {
+      return A.numElements;
+    }
+    proc resize(size: int) {
+      D = {0..#size};
+    }
+    proc count(val: eltType) {
+      return A.count(val);
+    }
+    iter these() {
+      for a in A do yield a;
+    }
+  }
+
+  class LoopSuiteRunInfo {
+    var host_name: string;
+
+    var num_loops: int;
+    var loop_names = new vector(string);
+
+    var num_loop_lengths: int;
+    var run_loop_length = new vector(bool);
+    var loop_length_names = new vector(string);
+
+    var num_suite_passes: int;
+    var loop_samp_frac: real;
+
+    var ref_loop_stat: LoopStat;
+
+    var loop_weights = new vector(real);
+
+    var num_loops_run = new vector(vector(int));
+    var tot_time = new vector(vector(real));
+    var fom_rel = new vector(vector(real));
+    var fom_rate = new vector(vector(real));
+
+    var cache_flush_data_len: int;
+    var cache_flush_data_dom: domain(1);
+    var cache_flush_data: [cache_flush_data_dom] real;
+    var cache_flush_data_sum: real;
+
+    var loop_test_stats_dom: domain(string);
+    var loop_test_stats: [loop_test_stats_dom] vector(LoopStat);
+
+    proc getLoopStats(loop_variant_name: string) {
+      return loop_test_stats[loop_variant_name];
+    }
+    proc addLoopStats(name: string) {
+      loop_test_stats_dom += name;
+      loop_test_stats[name] = new vector(LoopStat);
+    }
+  }
+
+  class LoopStat {
+    var loop_is_run: bool;
+    var loop_weight: real;
+
+    var loop_run_time = new vector(vector(real));
+    var loop_run_count = new vector(int);
+    var mean = new vector(real);
+    var std_dev = new vector(real);
+    var min = new vector(real);
+    var max = new vector(real);
+    var harm_mean = new vector(real);
+    var meanrel2ref = new vector(real);
+
+    var loop_length = new vector(int);
+    var samples_per_pass = new vector(int);
+
+    var loop_chksum = new vector(real);
+
+    proc LoopStat(num_loop_lengths: int) {
+      loop_is_run = false;
+      loop_weight = 0.0;
+
+      loop_run_time.resize(num_loop_lengths);
+      for i in 0..#num_loop_lengths do
+        loop_run_time[i] = new vector(real);
+      loop_run_count.resize(num_loop_lengths);
+      mean.resize(num_loop_lengths);
+      std_dev.resize(num_loop_lengths);
+      min.resize(num_loop_lengths);
+      max.resize(num_loop_lengths);
+      harm_mean.resize(num_loop_lengths);
+      meanrel2ref.resize(num_loop_lengths);
+
+      loop_length.resize(num_loop_lengths);
+      samples_per_pass.resize(num_loop_lengths);
+
+      loop_chksum.resize(num_loop_lengths);
+    }
+  }
+
+  record ADomain {
+    /*static*/ var loop_length_factor: real = 1.0;
```

Comment:
In the C reference version, it was a static field, so all instances of the structure share a single copy.  In Chapel we don't have a way to express that, but I noted that it was static originally in that comment.  As long as we initialize them all to the same value and don't change any it should be functionally equivalent.

I need to look at the variables named loop_length_factor a little more closely before it is correct to use anything other than the default of 1.0 and have it update everywhere it needs to correctly.  So far it hasn't been a problem because I haven't needed to change away from the default.